quickstart/kubernetes-guestbook.md

```diff
@@ -0,0 +1,373 @@
+                            // value: "env",
+                        }],
+                        ports: [{ containerPort: 6379 }],
+                    }],
+                },
+            },
+        },
+    });
+
+    // FRONTEND
+    let frontendLabels = { app: "guestbook", tier: "frontend" };
+    let frontendService = new k8s.core.v1.Service("frontend", {
+        metadata: {
+            name: "frontend",
+            labels: frontendLabels,
+        },
+        spec: {
+            // Comment or delete the following line if you want to use a LoadBalancer:
+            type: "NodePort",
+            // If your cluster supports it, uncomment the following to automatically create
+            // an external load-balanced IP for the frontend service.
+            // type: "LoadBalancer",
+            ports: [{ port: 80 }],
+            selector: frontendLabels,
+        },
+    });
+    let frontendDeployment = new k8s.apps.v1.Deployment("frontend", {
+        metadata: { name: "frontend" },
+        spec: {
+            selector: { matchLabels: frontendLabels },
+            replicas: 3,
+            template: {
+                metadata: { labels: frontendLabels },
+                spec: {
+                    containers: [{
+                        name: "php-redis",
+                        image: "gcr.io/google-samples/gb-frontend:v4",
+                        resources: {
+                            requests: {
+                                cpu: "100m",
+                                memory: "100Mi",
+                            },
+                        },
+                        env: [{
+                            name: "GET_HOSTS_FROM",
+                            value: "dns",
+                            // Using `GET_HOSTS_FROM=dns` requires your cluster to provide a dns service. As of
+                            // Kubernetes 1.3, DNS is a built-in service launched automatically. However, if the
+                            // cluster you are using does not have a built-in DNS service, you can instead access an
+                            // environment variable to find the master service's host. To do so, comment out the
+                            // 'value: "dns"' line above, and uncomment the line below:
+                            // value: "env",
+                        }],
+                        ports: [{ containerPort: 80 }],
+                    }],
+                },
+            },
+        },
+    });
+    ```
+
+    This is a lot of code, but it just creates three Kubernetes Services, each with an associated Deployment. The full
+    Kubernetes object model is available to us. Programming directly against the raw object model is a great way to ease
+    into using Pulumi, and gives us the full power of Kubernetes right away; however, we will also soon see that real
+    languages can help greatly simplify examples like this, thanks to the added abstraction of classes and functions.
```

Comment:
> however, we will also soon see that real languages can help greatly simplify examples like this, thanks to the added abstraction of classes and functions.

I don't see any followup on this point in the text here.  

Should we try to just use an example up front that is more Pulumi-natural, and avoid excusing the "lot of code" (highlighting instead how abstractions have helped us reduce repetition and capture common patterns and reuse)?

Reply:
Yeah, we need to build on this and add the aforementioned extension. I think it's fine to mention in passing just in case someone starts to think "hey, this just looks like YAML written in JavaScript."

Reply:
(and once we have it, we can hyperlink to it)